src/components/v5/common/ActionSidebar/partials/MultiSigSidebar/partials/MultiSigWidget/MultiSigWidget.tsx

```diff
@@ -23,8 +22,7 @@ const displayName =
   'v5.common.ActionSidebar.partials.MultiSig.partials.MultiSigWidget';
 
 interface MultiSigWidgetProps {
-  actionType: ColonyActionType;
-  multiSigData: ColonyMultiSig;
+  action: Omit<ColonyAction, 'multiSigData'> & { multiSigData: ColonyMultiSig };
```

Comment:
At this point, `multiSigId` is still defined as an optional field but looking at the changes you've made in `MultiSigSidebar.tsx`, you're passing in an action object which guarantees that its `multiSigId` field is not optional and is definitely a string. I think a better solution to make your approach more airtight is to declare a different type like:

```ts
type DefinitelyMultiSig = Omit<ColonyAction,'multiSigData' | 'multiSigId'> & { multiSigData: ColonyMultiSig; multiSigId: string };
```

Then you can slightly rewrite your `hasMultiSig` function as:
```ts
function hasMultiSig(action: ColonyAction): action is DefinitelyMultiSig {
  return !!action.multiSigData && !!action.multiSigId;
}
```

Then this `action` field as:
```ts
interface MultiSigWidgetProps {
  ...
  action: DefinitelyMultiSig;
  ...
}
```

P.S. `DefinitelyMultiSig` is written in jest. You can obviously give it a much better name 😂